src/entity-type-def-manager.ts

```diff
@@ -348,30 +344,76 @@ export class EntityTypeDefManager {
               nexusBuilder,
             ) as Nexus.core.AllInputTypes
             if (columnTypeName === 'String') {
-              singleOperandOperations.forEach(singleOperandOperation => {
-                t.field(`${column.propertyName}_${singleOperandOperation}`, {
-                  type: columnTypeName!,
-                })
+              t.field(column.propertyName, {
+                type: 'StringFilter',
               })
             } else if (columnTypeName === 'Int' || columnTypeName === 'Float') {
-              numberOperandOperations.forEach(numberOperandOperation => {
-                t.field(`${column.propertyName}_${numberOperandOperation}`, {
-                  type: columnTypeName!,
-                })
+              t.field(column.propertyName, {
+                type: 'IntFilter',
+              })
+            } else if (columnTypeName === 'ID') {
+              t.field(column.propertyName, {
+                type: 'IdFilter',
+              })
+            } else {
+              t.field(column.propertyName, {
+                type: columnTypeName,
               })
             }
+          })
+        },
+      }),
+    )
 
-            // Define ${arg}_${operand}: Value[]
-            multipleOperandOperations.forEach(multipleOperandOperation => {
-              t.field(`${column.propertyName}_${multipleOperandOperation}`, {
-                type: columnTypeName!,
-                list: true,
-              })
-            })
+    this.typesDictionary.push(typeName)
+    return typeName
+  }
 
-            t.field(column.propertyName, {
-              type: columnTypeName,
-            })
+  useWhereUniqueInputType(
+    entity: Function,
+    nexusBuilder: NexusSchemaBuilder,
+  ): Nexus.core.AllInputTypes {
+    const entityTypeName = getEntityTypeName(entity)
+    const typeName = namingStrategy.whereUniqueInputType(entityTypeName) as Nexus.core.AllInputTypes
+
+    if (this.typesDictionary.includes(typeName)) {
+      return typeName
+    }
+
+    const { columns: entityColumns, indices, uniques } = getConnection().getMetadata(entity)
+
+    // find indices columns, and for now only allow inices that registers only one column
+    const uniqueIndices = [
+      indices
+        .filter(index => index.isUnique)
+        .map(index => index.columns.map(column => column.propertyName)),
+      uniques.map(unique => unique.columns.map(column => column.propertyName)),
+    ].flat(10)
```

Comment:
The comment is misleading : this allows for all indexes AND @Column({unique : true})